test/test-jsdomparser.js

```diff
@@ -190,6 +195,19 @@ describe("Test JSDOM functionality", function() {
       }
     }
   });
+
+  it("should handle encoding HTML correctly", function() {
+    var baseStr = "<p>Hello, everyone &amp; all their friends, &lt;this&gt; is a &quot; test.</p>";
+    var doc = new JSDOMParser().parse(baseStr);
+    var p = doc.getElementsByTagName("p")[0];
+    expect("<p>" + p.innerHTML + "</p>").eql(baseStr);
+
+    p.textContent = p.textContent + " ";
+    expect("<p>" + p.innerHTML + "</p>").eql(baseStr.replace("</p>", " </p>"));
+
+    var doc = new JSDOMParser().parse("<p>&#32;&#x20;</p>");
+    expect(doc.getElementsByTagName("p")[0].textContent).eql("  ");
```

Comment:
I suggest we generally split tests containing many assertions into smaller ones, so when the suite fails we have a more precise idea what is exactly failing.